app/controllers/spree/subscriptions_controller.rb

```diff
@@ -23,25 +23,22 @@ def update
     end
 
     def cancel
```

Comment:
shouldn't we wrap this whole logic in a service so that we can execute same from console or anyplace as well?

Reply:
this service should be like other spree services i.e return true if all good else returns false, check those for reference